server/service/client.go

```diff
@@ -30,9 +32,29 @@ func NewClient(addr string, insecureSkipVerify bool) (*Client, error) {
 		return nil, errors.Wrap(err, "parsing URL")
 	}
 
+	rootCAPool, _ := x509.SystemCertPool()
+	if rootCA != "" {
+		// set up empty cert pool
+		rootCAPool = x509.NewCertPool()
+
+		// read in the root cert file specified in the context
+		certs, err := ioutil.ReadFile(rootCA)
+		if err != nil {
+			fmt.Printf("Failed to append %v to RootCAs: %v\n", rootCA, err)
```

Comment:
We need to propagate the error here, as otherwise we will continue with an empty cert chain. Please use `errors.Wrap` (many examples through this file).